autometrics/src/labels.rs

```diff
@@ -221,3 +223,22 @@ pub trait GetStaticStr {
     }
 }
 impl_trait_for_types!(GetStaticStr);
+
+/// Implementation detail to get enum variants to specify their own
+/// "result" label
+pub trait GetErrorLabel {
+    /// Return the value to use for the [result](RESULT_KEY) value in the reported metrics
+    fn __autometrics_get_error_label(&self) -> &'static str {
+        ERROR_KEY
+    }
+}
+impl_trait_for_types!(GetErrorLabel);
+
+/// Implementation detail to get enum variants to specify their own
+/// "result" label
+pub trait GetErrorLabelFromEnum {
```

Comment:
Should we make this trait public (and rename it to match the derive macro)?

Reply:
This one wasn't necessary actually, I thought that if I didn't have an extra trait, I would have "conflicting implementation for trait `GetErrorLabel`" errors from the derive macro. I just tried removing it and it still compiles fine.